s-ramp-ide/srampRepositoryBrowser/src/main/java/cz/muni/fi/srampRepositoryBrowser/manager/BrowserManagerImpl.java

```diff
@@ -0,0 +1,197 @@
+package cz.muni.fi.srampRepositoryBrowser.manager;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.oasis_open.docs.s_ramp.ns.s_ramp_v1.BaseArtifactType;
+import org.overlord.sramp.atom.err.SrampAtomException;
+import org.overlord.sramp.client.SrampAtomApiClient;
+import org.overlord.sramp.client.SrampClientException;
+import org.overlord.sramp.client.SrampClientQuery;
+import org.overlord.sramp.client.query.ArtifactSummary;
+import org.overlord.sramp.client.query.QueryResultSet;
+import org.overlord.sramp.common.ArtifactType;
+
+/**
+ * Browser manager interface impl.
+ * 
+ * @author Jan Bouska
+ * 
+ */
+
+public class BrowserManagerImpl implements BrowserManager {
+
+	private SrampAtomApiClient client;
+	private boolean isConected = false;
+
+	public static final Logger log = Logger.getLogger(BrowserManagerImpl.class
+			.getName());
+
+	public void setConnection(final String endpoint, final String username,
+			final String password) throws ServiceFailureException {
+		try {
+			client = new SrampAtomApiClient(endpoint, username, password, true);
+			isConected = true;
+			log.log(Level.INFO, "connected.");
+		} catch (SrampClientException e) {
+			log.log(Level.WARNING, "Problem with connecting", e);
+			throw new ServiceFailureException("Problem with connecting", e);
+		} catch (SrampAtomException e) {
+			log.log(Level.WARNING, "Problem with connecting", e);
+			throw new ServiceFailureException("Problem with connecting", e);
+		}
+	}
+
+	public boolean isConnected() {
+		return isConected;
+	}
+
+	public SrampClientQuery buildQuery(String query) {
+		return client.buildQuery(query);
+	}
+
+	public SrampClientQuery listAllArtifacts() {
+
+		return buildQuery("/s-ramp[@derived = 'false']");
+
+	}
+
+	public void uploadArtifact(IFile content, String name, String type,
+			Properties prop) throws ServiceFailureException
+
+	{
+		if (!content.exists()) {
+			log.log(Level.WARNING,
+					"Problem in importing the artifact into the S-RAMP repository - file doesn't exists.");
+			throw new ServiceFailureException(
+					"Problem in importing the artifact into the S-RAMP repository - file doesn't exists.");
+		}
+
+		if (name.isEmpty()) {
+			name = content.getName();
+
+		}
+
+		if (type.isEmpty()) {
```

Comment:
See the note on ArtifactTypeGuessingService.java.  This can be removed -- simply send null or empty string to the server if no type is available.  The server itself will automatically detect the type.